### PR TITLE
chore: Add summary for received filters in bloom gateway

### PR DESCRIPTION
**Pull request by salvacorts**
**What this PR does / why we need it**:
This PR addresses a TODO from https://github.com/grafana/loki/pull/12135 to add a metric to track the number of filters per request.

The comment suggests adding it to the query frontend but:
- I think this is mainly useful on the blooms path
- This will allow us to see if any of the gateways is receiving more filters than the rest.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Review by owen-d — COMMENTED** [review]
We probably want this in two places:
1) in the query-frontend (as I mentioned in the comment) that logs 1:1 with the original query. This gives one observation per query and ensures we measure queries that have zero filters which are never passed to the bloom-gws.
2) in the bloom-gws that measure an observation once per exectuion (post split+shard in the query frontend). 

Also, we should use histograms and not summaries, which gives us aggregation support. In general, histograms are what you want instead of summaries in most cases. [context](https://prometheus.io/docs/practices/histograms/#:~:text=The%20essential%20difference%20between%20summaries,the%20server%20side%20using%20the)

File: pkg/bloomgateway/metrics.go [hunk 1]
```diff
@@ -66,6 +67,13 @@ func newServerMetrics(registerer prometheus.Registerer, namespace, subsystem str
 			Help:      "Total amount of chunk refs filtered by bloom-gateway",
 			Buckets:   prometheus.ExponentialBucketsRange(1, 100e3, 10),
 		}),
+		receivedFilters: promauto.With(registerer).NewHistogram(prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: subsystem,
+			Name:      "request_filters",
+			Help:      "Number of filters per request.",
+			Buckets:   prometheus.LinearBuckets(0, 1, 20),
```
Comment (owen-d) [inline pkg/bloomgateway/metrics.go:75]: I don't think we'll need 20 1-width buckets. I've suggested an exponential increasing range up to 256
```suggestion
			// 1->256
			Buckets:   prometheus.ExponentialBuckets(1, 2, 8),
```

File: pkg/querier/queryrange/metrics.go [hunk 2]
```diff
@@ -35,6 +40,57 @@ func NewMetrics(registerer prometheus.Registerer, metricsNamespace string) *Metr
...
+	receivedFilters prometheus.Histogram
+}
+
+func NewMiddlewareQueryMetrics(registerer prometheus.Registerer, metricsNamespace string) *QueryMetrics {
+	return &QueryMetrics{
+		receivedFilters: promauto.With(registerer).NewHistogram(prometheus.HistogramOpts{
+			Namespace: metricsNamespace,
+			Name:      "query_frontend_query_filters",
+			Help:      "Number of filters per query.",
+			Buckets:   prometheus.LinearBuckets(0, 1, 20),
```
Comment (owen-d) [inline pkg/querier/queryrange/metrics.go:58]: Use the same buckets here

**Review by owen-d — APPROVED** [review]
approving, but please update the buckets. Thanks!

File: pkg/bloomgateway/metrics.go (hunk 1 above)
Comment (salvacorts) [reply pkg/bloomgateway/metrics.go]: Done

File: pkg/querier/queryrange/metrics.go (hunk 2 above)
Comment (salvacorts) [reply pkg/querier/queryrange/metrics.go]: Done

**Pull request merged**